### ISSUE TITLE
Recovering 3TB WDBFJK0030HBK-04 Drive

### ISSUE THREAD

**Issue by tittaenalg** [andlabs/reallymine#101]
I believe I am having a very similar issue to [#88](https://github.com/andlabs/reallymine/issues/88). Same lack of power and same asmedia chip on the SATA to USB board, so I do not believe the data is encrypted. Before browsing issues, I attempted to run the `reallymine getdek` command, but had no results after leaving it to run for about an hour, which would make sense if the drive isn't encrypted.

 I removed the disk from its original case, and have placed it in a new external enclosure. When I connect it to my Mac, the computer sees it but says `The disk you inserted was not readable by this computer` and gives me options to `Initialize...`, `Ignore`, or `Eject`. If I click `Ignore` and look in Disk Utility, I can see the drive.

Following the procedure from that thread, I performed the below commands.

Running `dd` on the Mac returns:
```
Password:
1+0 records in
1+0 records out
512 bytes transferred in 0.000282 secs (1816822 bytes/sec)
00000000  fa b8 00 10 8e d0 bc 00  b0 b8 00 00 8e d8 8e c0  |................|
00000010  fb be 00 7c bf 00 06 b9  00 02 f3 a4 ea 21 06 00  |...|.........!..|
00000020  00 be be 07 38 04 75 0b  83 c6 10 81 fe fe 07 75  |....8.u........u|
00000030  f3 eb 16 b4 02 b0 01 bb  00 7c b2 80 8a 74 01 8b  |.........|...t..|
00000040  4c 02 cd 13 ea 00 7c 00  00 eb fe 00 00 00 00 00  |L.....|.........|
00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000070  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000080  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000100  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000110  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000120  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000130  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000140  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000150  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000160  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000170  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000180  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000190  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001b0  00 00 00 00 00 00 00 00  38 f2 07 00 00 00 00 04  |........8.......|
000001c0  05 00 83 fe ff ff 00 01  00 00 00 f3 a9 2b 00 00  |.............+..|
000001d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 55 aa  |..............U.|
00000200
```

I then booted up a virtual instance of Ubuntu, connected the drive there, and ran `fdisk -l` with the following results for the relevant drive:
```
Disk /dev/sdb: 2.7 TiB, 3000559428096 bytes, 5860467633 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes
Disklabel type: dos
Disk identifier: 0x0007f238

Device     Boot Start       End   Sectors   Size Id Type
/dev/sdb1         256 732558335 732558080 349.3G 83 Linux
```

Running `dd` on `/dev/sdb` in Ubuntu resulted in:
```
00000000  fa b8 00 10 8e d0 bc 00  b0 b8 00 00 8e d8 8e c0  |................|
00000010  fb be 00 7c bf 00 06 b9  00 02 f3 a4 ea 21 06 00  |...|.........!..|
00000020  00 be be 07 38 04 75 0b  83 c6 10 81 fe fe 07 75  |....8.u........u|
00000030  f3 eb 16 b4 02 b0 01 bb  00 7c b2 80 8a 74 01 8b  |.........|...t..|
00000040  4c 02 cd 13 ea 00 7c 00  00 eb fe 00 00 00 00 00  |L.....|.........|
1+0 records in
1+0 records out
512 bytes copied, 0.00151764 s, 337 kB/s
00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000070  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000080  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000100  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000110  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000120  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000130  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000140  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000150  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000160  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000170  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000180  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000190  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001b0  00 00 00 00 00 00 00 00  38 f2 07 00 00 00 00 04  |........8.......|
000001c0  05 00 83 fe ff ff 00 01  00 00 00 f3 a9 2b 00 00  |.............+..|
000001d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 55 aa  |..............U.|
00000200
```

And running `fdisk -l` on the above dump results in:
```
Disk ubuntuAttached.bin: 2.5 KiB, 2560 bytes, 5 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
```

I'm not seeing a description of blocks above, so I am unsure if my issue is exactly the same as the related issue, or this is something different. I am interested in attempting the `sudo fdisk /dev/sdX` command recommended in the other issue to write a new partition table, but want to be sure that is the correct action to perform before doing so.

If you need any further information, please let me know. I appreciate the help!

**Comment by themaddoctor** [comment]
Yes, it looks like the same issue as #88.

**Comment by isidroco** [comment]
Just in case someone's searching for same problem: This drive ASM chip doesn't encrypt but uses non standard 4k sectors. Ideally work with an image (I used HDDRawCopy2.5Portable)  but my HD had bad sectors at the very end so I kept partial image just in case and procceed to mount WD on a virtual VMware Lubuntu taken from: https://www.osboxes.org/lubuntu   actual NTFS partition starts at offsett: 100000h and has 4k sectors. Windows sees a RAW partition, don't touch it or format. Just go to VMware virtual machine settings, hardware tab, Add button, Hard Disk, SATA, use a physical disc. You must select entire disc, but first choose partitions for checking you are not selecting your own OS disc (in my case disc 0 was the HD to recover). 
Boot linux, open a command Terminal (Ctrl + T) and do the following:
ensure ReadOnly:  sudo blockdev --setro /dev/sdb
4k sectors mount, 100000h offset:  sudo losetup -f --show -r -b 4096 -o 1048576 /dev/sdb
mount ntfs with no recovery:  sudo mount -t ntfs-3g -o ro,norecover /dev/loop0 /mnt
verify that is consistent looking at root dir names:  ls /mnt
**[this may not be necessary] make shared dir:  sudo mkdir -p /home/$USER/mariana
** link it to drive:  sudo mount --bind /mnt /home/$USER/mariana
** set permissions:  sudo chmod 755 /home/$USER/mariana
net install:  sudo apt install samba
add [mariana] to samba settings:  sudo nano /etc/samba/smb.conf
[mariana]
   path = /mnt
   read only = yes
   guest ok = yes
(Save and exit)
Restart samba:  sudo systemctl restart smbd

After that, my drive was readable from windows using network, so I added another drive to backup all data.
